common/src/main/java/io/confluent/connect/storage/common/StorageCommonConfig.java

```diff
@@ -52,17 +56,23 @@
   public static final String FILE_DELIM_DEFAULT = "+";
   public static final String FILE_DELIM_DISPLAY = "File Delimiter";
 
-  protected static final ConfigDef CONFIG_DEF = new ConfigDef();
-  public static final GenericRecommender STORAGE_CLASS_RECOMMENDER =
-      new GenericRecommender();
+  public static ConfigDef newConfigDef(Collection<GenericRecommender> recommenders) {
```

Comment:
First of all, why is this not just `Recommender`? Also, why not just pass the recommender as a single named method argument rather than in a collection? 

Same questions for the other two config classes.

Reply:
1) That's valid now. I started thinking that values could be added at the base config, but there's no reason now. Will change to `ConfigDef.Recommender`
2) The fact that we only need one recommender per base config is purely incidental. I figured it's more extensible to account for multiple from the beginning. Not crazy about how this is written implying an order but feels good enough for now.

Reply:
These methods are static, so they can't be overloaded or overwritten. The use of `Collection<Recommender>` seems really complicated, and I'd prefer we just use a single simple `Recommender` parameter.

Reply:
These methods are not supposed to be overloaded. They are called from within the class or with the classname anyways. 

As I mentioned in the other repo too, with already a few class type properties and a few dozen properties overall, adding every recommender as a separate argument is not an obvious scalable solution. Map would be another option. 

Anyway, we'll cross this bridge when we come to it. Adding as simple argument at the moment.